---
title: 'Mapping vs core genome alignment'
date: 2017-04-24
permalink: /posts/2017/04/blog-post-1/
tags:
  - first post
  - genome mapping
  - bioinformatics
  - core genome alignment
  - population biology
---

Suppose we sequenced > 1000 strains of a single species. There are a number of different ways to compare the data, but at one point you might want to have a multiple sequence alignment of all strains. I was wondering if mapping reads against a reference would give comparable results to aligning the core genome using a reference?

Aligning core or mapping all reads?
======

<!---
A multiple sequence alignment can be used as input for further analysis such as mapping epistasic links like in the paper by [Skwark et al.](http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006508a). To achieve this from a large number of sequenced, closely related strains there are basically two options:

  - Mapping all reads against a reference genome, or
  - Assembling contigs and aligning the core genome

There are a lot of options to map reads against a reference genome, e.g. with [SMALT](http://www.sanger.ac.uk/science/tools/smalt-0), [Snippy](https://github.com/tseemann/snippy), or plain-and-simple with bwa, or, my recent favorite, [NASP](https://tgennorth.github.io/NASP/usage.html#nasp). The mapping itself doesn't pose too many problems (except of course for repeats and the like) if the reference is well-chosen. The consensus calling, however, 

You can have many headings
======

Aren't headings cool?
------
--->
