---
title: 'Popularity of SNP pipelines'
date: 2017-10-16
permalink: /posts/2017-10-16-blog-post-3
tags:
  - SNP pipelines
  - bioinformatics
  - microbial
  - twitter
---

Recently we got back a round of reviews on a small review paper. One specific comment made me think. Reviewer 2 disagreed with my choice of examples for SNP pipelines. The topic of the review is not pipelines so the intention was not to list them all, just one or two examples. He suggested to omit the name of the pipeline that was unpublished and name a few other published ones instead.

Well. 

In my mind, software publications are not that interesting. Sure, the software in the publication is the winner of the performed benchmark, something we could call the software publication benchmark bias. But the real deal, like, Does it install on my system? Is it applicable to my dataset? is usually not judged by the journal, editor or reviewer. Of course software publications can act as announcements but I hear more often about new tools on conferences (and then especially posters and showcases), twitter, or maybe bioRxiv.

Therefore I do not place a whole lot of worth on software publications. It's nice to have a reference but I do not have any problems of referencing a github page instead.

But the reviewer has a point. I arbitrarily picked a few examples without having an idea what is really popular among microbiology laboratories. To get to the ground of it I set up a twitter poll, initially to get a feeling on what is used, and then a second survey where I tried to quantify the popularity of pipelines a bit more.

A total of 66 responses were recorded on the survey, with 50% of respondents in medical microbiology. Here is the (very approximate) [map](https://drive.google.com/open?id=1X-szvMXaCjEMxcprRRvQqqrymk0&usp=sharing). Many people (24/66) indicated that they use several pipelines. 


<img src="https://aschuerch.github.io/images/201710_Pipechart.png" width="750">


In this survey, the most popular pipeline was Snippy with 19 mentions before custom pipelines which are used by 17. Also very popular is the use of ParSNP on assembled genomes with 12 votes. This is slightly different than the earlier [Twitter poll](https://twitter.com/AnitaSchurch/status/910816027902201856) where 42% of 36 votes claimed to run a custom pipeline. But the bottom-line is: There are 66 users and 21 different pipelines and that's even without the custom pipelines.

Of course, the population that took the survey is highly biased. The number of GUI pipelines for example is extremely low, which does not match what I am seeing, at least in European laboratories. But I probably reached more bioinformatics-minded people than purely lab-based people.

Thanks everyone who participated in the survey. And stay tuned if you wonder if the mention of the unpublished pipeline makes it through the revisions.


