---
title: 'Plasmid genomics: gplas'
date: 2020-01-10
permalink: /posts/2020-01-06-blog-post-7
tags:
  - bioinformatics
  - gplas
  - plasmid
  - chromosome
  - assembly graph
  - E. coli
  - E. faecium
  - K. pneumoniae
  - podcast
---

Update 2020-04-11: Now published in [Bioinformatics](https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btaa233/5818483?guestAccessKey=dbe63461-12ea-4c06-8007-e60fe41faacd)

gplas is a software tool to find different plasmids in short read sequenced bacterial genomes. 
It bins contigs that are predicted as plasmids into separate entities, corresponding to different plasmids. 
To do that, it uses kmer information to determine plasmid nodes, then it makes use of the nformation in the assembly graph, 
i.e. the connectivity of the nodes and finally it takes the coverage of the separate nodes into account. 
It works best in combination with the prediction of mlplasmids, thus with species-specific models for 
classification, but it can also use plasflow which has a species-independent model for classification.

The preprint is available at [biorXiv](https://www.biorxiv.org/content/10.1101/835900v1) and the algorithm can be 
retrieved from [Sergio Arredondo's gitlab repo](https://gitlab.com/sirarredondo/gplas.git)

Want to know more about it? There is a podcast about plasmid genomics, gplas and mlplasmids at 
the [bioinformatics chat](bioinformatics.chat/plasmids)
