---
title: 'vanA-type vancomycin-resistance dissemination'
date: 2020-07-22
permalink: /posts/2020-07-22-blog-post-8
tags:
  - bioinformatics
  - gplas
  - mplasmids
  - plasmid
  - chromosome
  - assembly graph
  - network analysis
  - E. faecium
---

Mobile genetic elements can be nested, like russian dolls. This is also the case for the *vanA* gene cluster in *Enterococcus faecium*, 
which is located on a Tn*1546* transposon, which, in turn, is often located on plasmids.

<img src="https://aschuerch.github.io/images/noun_Matryoshka_182630.png" width="250">

Entangling this net is difficult without long read sequences. However, we attempted to reconstruct the entire AMR nested genetic complex 
using only short-read WGS data in a dataset of vancomycin-resistant *E. faecium* isolates from Dutch hospitals from 2012 to 2015. 
We compared a core-genome phylogeny with transposon typing data and a plasmid network.
This plasmid network was enabled by machine learning and graph-based techniques coupled with a network analysis of the shared plasmid k-mer content 
It allowed us to elucidate not only clonal relatedness but also plasmid and transposon level dissemination of AMR. And while we found that most *vanA* 
spread in our data set 
was dominated by clonal dissemination, there were clear pockets of plasmids or transposon-dominated spread.

Our study is now available as a [preprint](https://). 

